xwiki-platform-core/xwiki-platform-template/xwiki-platform-template-api/src/main/java/org/xwiki/template/helpers/StringTemplate.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.template.helpers;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.rendering.syntax.Syntax;
+import org.xwiki.template.Template;
+import org.xwiki.template.TemplateContent;
+import org.xwiki.text.StringUtils;
+
+/**
+ * Allows the creation of a template directly by specifying its content as a string in the constructor.
+ *
+ * @version $Id$
+ * @since 9.5RC1
+ */
+public class StringTemplate implements Template
+{
+    private class StringTemplateContent implements TemplateContent
```

Comment:
Hmm actually it's not really how I was seeing it. It should behave like others TemplateContent implementations and let the template itself choose options like the syntax, etc. Most of that is taken into account by DefaultTemplateContent. So instead of a public StringTemplate class you should probably provide a TemplateManager#createStringTemplate(String content, DocumentReference author) which will reuse DefaultTemplateContent behind the scene.